### PR TITLE
feat: add support for latest FrSky receivers

### PR DESCRIPTION
**Pull request by 3djc**
This is a port from https://github.com/opentx/opentx/commit/29dd63af8bf7217033e1b5a553db0ccb9454546a

### CODE REVIEW

File: radio/src/telemetry/frsky_pxx2.cpp [hunk 1]
```diff
@@ -39,8 +39,8 @@ static const char * const PXX2ModulesNames[] = {
   "ISRM-PRO",
   "ISRM-S",
   "R9M",
-  "R9MLite",
-  "R9MLite-PRO",
+  "R9M Lit",
```
Comment (raphaelcoeffic) [inline radio/src/telemetry/frsky_pxx2.cpp:42]: Can I have one more `"e"` please?

**Comment by raphaelcoeffic** [comment]
I believe this obsoletes #4045.

**Review by raphaelcoeffic — APPROVED** [review]

**Comment by raphaelcoeffic** [comment]
Tested with a `SR10+` and `X10 Express`:
- [x] SBUS out
- [x] SBUS 24
- [x] S.PORT

**Comment by rburrow87** [comment]
I have an `Archer Plus SR10+`, `Archer Plus RS`, and `Archer Plus RS Mini`. 

With my X9D+2019, the list of options for the pin outs stop at CH24 on the SR10+.

With my X12S w/ISRM, I can see the S.PORT, SBUS out, and FBUS options on the SR10+. Everything seems functional with the X12S and all 3 receivers.

**Comment by raphaelcoeffic** [comment]
> With my X12S w/ISRM, I can see the S.PORT, SBUS out, and FBUS options on the SR10+.

You should be able to see "SBUS in" on the first pin as well. At least on the SR10+, not sure the others have that capability.

> Everything seems functional with the X12S and all 3 receivers.

Could you please test if "SBUS in" and "FBUS" are indeed woking?

**Comment by rburrow87** [comment]
I just tested CH1 set to `SBus in` with another RX's SBus out connected to it and it does work!

But, it does not look like FBUS works. I can select FBUS and it saves it, but there is no output. This applies to all 3 receivers. I can re-bind them to my Tandem radio without resetting the RX and FBUS will work there.

I tried S.Port and that seems to work fine. I also tried F.Port on the two Archer Plus RS receivers that also works.

**Comment by raphaelcoeffic** [comment]
> But, it does not look like FBUS works. I can select FBUS and it saves it, but there is no output. This applies to all 3 receivers. I can re-bind them to my Tandem radio without resetting the RX and FBUS will work there.

That probably means we need to send some unknown bit at some point.... Let's see what can be done.

**Comment by raphaelcoeffic** [comment]
@rburrow87 strangely, I see some traffic at ~460kbps. That isn't F.PORT (which is 115kbps) nor SBUS (100kbps) nor S.PORT (57.6kbps).

**Comment by mha1** [comment]
ELRS debug works at 460800 ...

**Comment by raphaelcoeffic** [comment]
@mha1 offtopic! These are really some FBUS frames I'm seeing. This is on a Archer PLUS receiver (SR10+).

[FBUS+SBUS.zip](https://github.com/EdgeTX/edgetx/files/12735527/FBUS%2BSBUS.zip)

The only strange thing is that the length announced is `35` bytes, instead of `24` or `32`, as it searched for in the iNav code (see [here](https://github.com/iNavFlight/inav/blob/master/src/main/rx/fport2.c#L254)).

File: radio/src/gui/colorlcd/access_settings.cpp [hunk 2]
```diff
@@ -710,24 +710,107 @@ void RxOptions::checkEvents()
 #define CH_ENABLE_SPORT 4
 #define CH_ENABLE_SBUS  5
 
+#define CH_MAP_SBUS_IN  (1 << 5) /* 0x20 */
```
Comment (bsongis) [inline radio/src/gui/colorlcd/access_settings.cpp:713]: 0b10100000

File: radio/src/gui/colorlcd/access_settings.cpp [hunk 3]
```diff
@@ -710,24 +710,107 @@ void RxOptions::checkEvents()
...
+#define CH_MAP_SPORT    (1 << 6) /* 0x40 */
+#define CH_MAP_SBUS_OUT (1 << 7) /* 0x80 */
+#define CH_MAP_FBUS     (CH_MAP_SPORT | CH_MAP_SBUS_OUT) /* 0xC0 */
```
Comment (bsongis) [inline radio/src/gui/colorlcd/access_settings.cpp:716]: 0b11000000

File: radio/src/gui/colorlcd/access_settings.cpp [hunk 4]
```diff
@@ -710,24 +710,107 @@ void RxOptions::checkEvents()
 #define CH_ENABLE_SPORT 4
 #define CH_ENABLE_SBUS  5
 
+#define CH_MAP_SBUS_IN  (1 << 5) /* 0x20 */
+#define CH_MAP_SPORT    (1 << 6) /* 0x40 */
+#define CH_MAP_SBUS_OUT (1 << 7) /* 0x80 */
```
Comment (bsongis) [inline radio/src/gui/colorlcd/access_settings.cpp:715]: 0b10000000

File: radio/src/gui/colorlcd/access_settings.cpp [hunk 5]
```diff
@@ -710,24 +710,107 @@ void RxOptions::checkEvents()
 #define CH_ENABLE_SPORT 4
 #define CH_ENABLE_SBUS  5
 
+#define CH_MAP_SBUS_IN  (1 << 5) /* 0x20 */
+#define CH_MAP_SPORT    (1 << 6) /* 0x40 */
```
Comment (bsongis) [inline radio/src/gui/colorlcd/access_settings.cpp:714]: 0b01000000

**Comment by 3djc** [comment]
@rburrow87 , could you test X9D+, think it should be ok now

**Comment by bsongis** [comment]
It won't work if the constants are not fixed

**Comment by rburrow87** [comment]
> @rburrow87 would you mind re-testing FBUS and set the number of channels to `16` (in "Channel Range")? Also, could you please tell me what you used to test with that was hooked to FBUS?

Well that was it! Maybe it should be defaulting to the more standard 16 channels instead of 24? I did not realize there was FBUS-16ch and FBUS-24ch.

I was testing FBUS with a Spirit GT heli FBL, one of the more modern FBL.

> @rburrow87 , could you test X9D+ 2019, think it should be ok now

I'm able to see and use the extra options on Pin1, but the rest of the pins still stop at CH16/24.

FBUS works here too after changing the channel range to 16.

**Comment by raphaelcoeffic** [comment]
> Maybe it should be defaulting to the more standard 16 channels instead of 24? I did not realize there was FBUS-16ch and FBUS-24ch.

Might be worth considering, indeed.

> I was testing FBUS with a Spirit GT heli FBL, one of the more modern FBL.

Ok, it probably only supports 16 channels. ArduPilot seems to support 24 channels as well.

> > @rburrow87 , could you test X9D+ 2019, think it should be ok now
> I'm able to see and use the extra options on Pin1, but the rest of the pins still stop at CH16/24.

This should be fixed now.

**Comment by rburrow87** [comment]
> > >  @rburrow87 , could you test X9D+ 2019, think it should be ok now
> >
> >  I'm able to see and use the extra options on Pin1, but the rest of the pins still stop at CH16/24.
> 
> This should be fixed now.

Looks good now! I tested with both my X9D+2019 and X12S w/ISRM.

Though it seems like there's an issue with the ISRM-S-X10S in my X12S that I missed in the first round of testing since I had it muted. I get constant telemetry lost/recovered with all 3 receivers, plus a regular Archer RS. Is this already known? I can submit a bug report if not. The issue goes away with ETX 2.9.0, and I don't see an issue with the X9D+2019 with 2.9.0 or this test build.

**Comment by rburrow87** [comment]
Doh, just noticed that when using an Archer RS, Archer Plus RS, or Archer Plus RSmini, I can no longer choose F.Port. The X9D+2019 only shows S.PORT and FBUS, but the X12S shows S.PORT twice and FBUS.

And this X12S telemetry issue seems like a larger issue as I just noticed that with an external ELRS module, the connection speed in status jumps all over the place instead of staying fixed 500hz when using a 500hz mode. I will do some more testing and submit a bug report for that TX.

**Comment by raphaelcoeffic** [comment]
> Doh, just noticed that when using an Archer RS, Archer Plus RS, or Archer Plus RSmini, I can no longer choose F.Port. The X9D+2019 only shows S.PORT and FBUS, but the X12S shows S.PORT twice and FBUS.

There was an error in the strings, should be fixed now.

Regarding the "telem lost/recovered" issue, I can see it as well on the test X10 Express with internal ISRM. Do I understand properly that you see this issue only the X12S, and not on the X9D+, right?

**Comment by rburrow87** [comment]
> > Doh, just noticed that when using an Archer RS, Archer Plus RS, or Archer Plus RSmini, I can no longer choose F.Port. The X9D+2019 only shows S.PORT and FBUS, but the X12S shows S.PORT twice and FBUS.
> 
> There was an error in the strings, should be fixed now.

I think it's all good now! Checked both the X9D+2019 and X12s.

> Regarding the "telem lost/recovered" issue, I can see it as well on the test X10 Express with internal ISRM. Do I understand properly that you see this issue only the X12S, and not on the X9D+, right?
> 
> _Edit: this is definitely not related to this PR, I can see it on `main` as well._ This PR seems to fix it for me: #4038

Correct, only an issue with the X12S. X9D+2019 seems solid. And that PR fixes it for me too!

**Comment by pfeerick** [comment]
I can only try out the 212x64 side of things since my only ACCESS transmitter is the X9D+2019, but this apperas to be working fine with an Archer Plus SR10+ and Archer Plus GR6, as well as the older Archer GR8. However, should I not be able to OTA update both of these RX like I can the GR8? As for the both the Archer Plus RXs I'm only getting the option to update via SPort. Just checking if it's user error or if something still needs to be implemented.

**Comment by 3djc** [comment]
Will try to investigate (but without any access gear, it's a bit harder)

File: radio/src/translations/untranslated.h
Comment (Fortuna599) [inline radio/src/translations/untranslated.h:1]: Is this code to work the SR10+ stabilizer and configure it? I'm looking for how to calibrate and adjust the SR10+ stabilizer on the Edgetx 2.9.2 TX16s, can you help me?

Comment (pfeerick) [reply radio/src/translations/untranslated.h:1]: Asked and answered in #4568. This was not quite the right place to ask anyway.

**Pull request merged**